File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> h = new HashMap<Integer, Integer>();

        for (int i=0; i<nums.length; i++){
            int complement = target-nums[i];
            if (h.containsKey(complement)){
                int[] ans = {i, h.get(complement)};
                return ans;
            }
            h.put(nums[i],i);
        } 
        return null;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map<int,int> m;

       for (int i=0;i<nums.size();i++){
           int complement = target-nums[i];
           if (m.find(complement)!=m.end()){
               vector<int> sol;
               sol.push_back(i);
               sol.push_back(m.find(complement)->second);
               return sol;
           }
           m[nums[i]]=i;
       }
       return vector<int>();
    }
};